packages/compass-utils/src/ipc.ts

```diff
@@ -0,0 +1,160 @@
+import { ipcMain, ipcRenderer, BrowserWindow } from 'electron';
+
+type SerializedError = { $$error: Error & { statusCode?: number } };
+
+// We are serializing errors to get a better error shape on the other end, ipc
+// will only preserve message from the original error. See https://github.com/electron/electron/issues/24427
+function serializeErrorForIpc(err: any): SerializedError {
+  return {
+    $$error: {
+      name: err.name,
+      message: err.message,
+      statusCode: err.status,
+      stack: err.stack,
+    },
+  };
+}
+
+function deserializeErrorFromIpc({ $$error: err }: SerializedError) {
+  const e = new Error(err.message);
+  e.name = err.name;
+  e.stack = err.stack;
+  (e as any).stausCode = err.statusCode;
+  return e;
+}
+
+function isSerializedError(err: any): err is { $$error: Error } {
+  return err !== null && typeof err === 'object' && !!err.$$error;
+}
+
+// Exported for testing purposes
```

Comment:
Then it’s probably a good idea to not do `export * from './ipc';` in index.ts and instead list the set of exports explicitly, I’d say :slightly_smiling_face: